core/services/ardupilot_manager/mavlink_proxy/Endpoint.py

```diff
@@ -1,76 +1,82 @@
-
-        for stringType, endpointType in self._connTypeMap:
-            if args[0] in stringType:
-                self.connType = endpointType
-                break
-        else:
-            self.connType = EndpointType.Undefined
 
-        self.place = args[1]
-        self.argument = args[2] if len(args) == 3 else None
+class EndpointType(str, Enum):
+    UDPServer = "udpin"
+    UDPClient = "udpout"
+    TCP = "tcp"
+    Serial = "serial"
+    File = "file"
 
-        return self
 
-    def from_dict(self, pattern: Dict[str, str]) -> "Endpoint":
-        if not {"connType", "place"}.issubset(pattern.keys()):
-            diff = {"connType", "place"}.difference(pattern.keys())
-            raise RuntimeError(f"Wrong format for endpoint creation, missing the following: {diff}.")
-
-        for stringType, endpointType in self._connTypeMap:
-            if pattern["connType"] in stringType:
-                self.connType = endpointType
-                break
-        else:
-            self.connType = EndpointType.Undefined
-
-        self.place = pattern["place"]
-        self.argument = pattern["argument"] if pattern["argument"] else None
-
-        return self
+@dataclasses.dataclass
+class Endpoint:
+    connection_type: str
+    place: str
+    argument: Optional[int] = None
+
+    @root_validator
+    @classmethod
+    def is_mavlink_endpoint(cls: Type["Endpoint"], values: Any) -> Any:
+        connection_type, place, argument = (values.get("connection_type"), values.get("place"), values.get("argument"))
+        if connection_type in [EndpointType.UDPServer, EndpointType.UDPClient, EndpointType.TCP]:
+            if validators.domain(place) or validators.ipv4(place) or validators.ipv6(place):
+                if argument in range(1, 65536):
+                    return values
+                raise ValueError(f"Ports must be in the range 1:65535. Received {argument}.")
+            raise ValueError(f"Invalid network address: {place}")
+
+        if connection_type == EndpointType.Serial.value:
+            if is_valid_filepath(place):
+                if argument in VALID_SERIAL_BAUDRATES:
+                    return values
+                raise ValueError(
+                    f"Invalid serial baudrate. Valid option are{VALID_SERIAL_BAUDRATES}. Received {argument}."
+                )
+            raise ValueError(f"Bad serial address: {place}. Make sure to use an absolute path.")
+
+        if connection_type == EndpointType.File.value:
+            if is_valid_filepath(place):
+                if argument is None:
+                    return values
+                raise ValueError(f"File endpoint has should have no argument. Received {argument}.")
+            raise ValueError(f"Bad file path: {place}. Make sure to use an absolute path.")
+        raise ValueError(
+            f"Invalid connection_type: {connection_type}. Valid types are: udpin, udpout, tcp, serial and file."
```

Comment:
valid types should be populated from enum and not hardcoded, doing like that makes it harder to maintain and also duplicates the information unnecessary.